src/gui/widgets/window.cpp

```diff
@@ -1307,8 +1307,12 @@ void window::signal_handler_sdl_key_down(const event::ui_event event,
 		}
 	}
 	if(!enter_disabled_ && (key == SDLK_KP_ENTER || key == SDLK_RETURN)) {
-		set_retval(retval::OK);
-		handled = true;
+		if (mod & KMOD_SHIFT) {
```

Comment:
What exactly is this doing? Why is shift special? Why not alt or control?

Reply:
> What exactly is this doing?
Allowing Shift+Enter even when enter is disabled.
> Why is shift special? Why not alt or control?
Mostly because Shift+Enter is used in Discord for linebreak. I can change this to Ctrl+Enter/Alt+Enter if you think that is more logical. Maybe I should let Enter used whenever it is pressed with a modifier? Something like :
`if (mod & KMOD_SHIFT||mod & KMOD_CTRL||mod & KMOD_ALT||mod & KMOD_GUI) {`

Reply:
This is the place that handles the "default button" functionality, right? I think it's best if the default button is only triggered by enter with no modifiers, so like `if(mod == 0) { set_retval(OK); }`.